49594 ddl unit tests

kjduensing opened the pull request:
## Summary

Add unit tests for Decision Letter Download feature

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#49594

## Testing done

This is only unit tests


## What areas of the site does it impact?
CI

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [-]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

jerekshoe committed:
Adding first unit test

jerekshoe committed:
Merged with main

jerekshoe committed:
Adding unit tests

jerekshoe committed:
Merge branch 'main' into 49594-ddl-unit-tests

kjduensing committed:
Merge branch 'main' of github.com:department-of-veterans-affairs/vets-website into 49594-ddl-unit-tests

kjduensing committed:
Add unit tests for YourClaimLetters

kjduensing committed:
Merge branch '49594-ddl-unit-tests' of github.com:department-of-veterans-affairs/vets-website into 49594-ddl-unit-tests

Mottie commented on src/applications/claims-status/containers/errorComponents/UnauthenticatedContent.jsx:
```diff
@@ -0,0 +1,13 @@
+      Something went wrong on our end. Please double check the URL and make sure
+      you are signed in.
+    </div>
+  </div>
```
I feel like all these error messages should be inside a `va-alert`. I haven't seen the design, but I feel like it makes sense.

jerekshoe replied:
I personally agree with you, but this is the design that was approved:
https://www.sketch.com/s/8b025901-295d-4929-95cc-8dbeb28454b3/a/dlL1L52

jerekshoe replied:
![Screen Shot 2023-01-04 at 12 03 26 PM](https://user-images.githubusercontent.com/13838621/210620299-50131008-1416-4b58-8796-99b328e5bf4f.png)

jerekshoe commented:
@kjduensing I'm thinking we should turn the YourClaimLetters container into a folder and put all of the associated components (ClaimLettersList, ClaimLettersListItem, and the error components) in it. We can take the code from `YourClaimLetters.jsx` and put it in the folder under an `index.jsx` file as well

kjduensing committed:
Reorganize components of YourClaimLetters

kjduensing committed:
Remove lint disable

kjduensing committed:
Add unit tests

Added comment explaining why we were leaving in commented code. Also
added a test for checking that the loading element appears if the
feature flag is loading.

Note that the coverage is a little wonky - the linter requires (rightly)
an ReactWrapper.unmount() for every enzyme mount() call, but for some
reason, calling `unmount()` causes coverage to suffer.

kjduensing committed:
Fixed false positives

Previously was using enzyme to check for presence of components. This
led to a bunch of false positives. If I removed the render of a
subcomponent in the parent component, the test would still pass. I
figured this was due to some async stuff in the `useEffect` hook, so I
read up a little bit and realized that the react-testing-library works
well with async things. I switched to use react-testing-library and am
now confident that things are being tested correctly.
